src/celeritas/em/distribution/MuBBEnergyDistribution.hh

```diff
@@ -0,0 +1,169 @@
+    real_type max_secondary_energy_;
+    // Whether to apply the radiative correction
+    bool use_rad_correction_;
+    // Envelope distribution for rejection sampling
+    real_type envelope_;
+
+    //// CONSTANTS ////
+
+    //! Incident energy above which the radiative correction is applied [MeV]
+    static CELER_CONSTEXPR_FUNCTION Energy rad_correction_limit()
+    {
+        return units::MevEnergy{250};  //!< 250 MeV
+    }
+
+    //! Lower limit of radiative correction integral [MeV]
+    static CELER_CONSTEXPR_FUNCTION Energy kin_energy_limit()
+    {
+        // This is the lower limit of the energy transfer from the incident
+        // muon to the delta ray and radiative gammas
+        return units::MevEnergy{0.1};  //!< 100 keV
+    }
+
+    //! Fine structure constant over two pi
+    static CELER_CONSTEXPR_FUNCTION real_type alpha_over_twopi()
+    {
+        return constants::alpha_fine_structure / (2 * constants::pi);
+    }
+
+    //// HELPER FUNCTIONS ////
+
+    inline CELER_FUNCTION real_type calc_envelope_distribution() const;
+};
+
+//---------------------------------------------------------------------------//
+// INLINE DEFINITIONS
+//---------------------------------------------------------------------------//
+/*!
+ * Construct with incident and exiting particle data.
+ */
+CELER_FUNCTION
+MuBBEnergyDistribution::MuBBEnergyDistribution(Energy inc_energy,
+                                               Mass inc_mass,
+                                               real_type beta_sq,
+                                               Mass electron_mass,
+                                               Energy electron_cutoff,
+                                               Energy max_secondary_energy)
+    : inc_energy_(value_as<Energy>(inc_energy))
+    , inc_mass_(value_as<Mass>(inc_mass))
+    , total_energy_(inc_energy_ + inc_mass_)
+    , beta_sq_(beta_sq)
+    , electron_mass_(value_as<Mass>(electron_mass))
+    , electron_cutoff_(value_as<Energy>(electron_cutoff))
+    , max_secondary_energy_(value_as<Energy>(max_secondary_energy))
+    , use_rad_correction_(
+          inc_energy_ > value_as<Energy>(rad_correction_limit())
+          && max_secondary_energy_ > value_as<Energy>(kin_energy_limit()))
+    , envelope_(this->calc_envelope_distribution())
+{
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Sample secondary electron energy.
+ */
+template<class Engine>
+CELER_FUNCTION auto MuBBEnergyDistribution::operator()(Engine& rng) -> Energy
+{
+    real_type energy;
+    real_type target;
+    do
+    {
+        energy = electron_cutoff_ * max_secondary_energy_
+                 / UniformRealDistribution(electron_cutoff_,
+                                           max_secondary_energy_)(rng);
+        target = 1 - beta_sq_ * energy / max_secondary_energy_
+                 + ipow<2>(energy) / (2 * ipow<2>(total_energy_));
```

Comment:
```suggestion
                 + real_type{0.5} * ipow<2>(energy / total_energy_);
```
Save a division and a multiplication 😉 (probably decreases our FLOP/s performance 😄)